beacon-chain/forkchoice/protoarray/optimistic_sync_test.go

```diff
@@ -0,0 +1,279 @@
+package protoarray
+
+import (
+	"context"
+	"testing"
+
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/config/params"
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+	"github.com/prysmaticlabs/prysm/testing/require"
+)
+
+// We test the algorithm to check the optimistic status of a node. The
+// status for this test is the following branching diagram
+//
+//                       -- E -- F
+//                      /
+//                  -- C -- D
+//                 /
+// 0 -- 1 -- A -- B      -- J -- K
+//                 \    /
+//                  -- G -- H -- I
+//
+// Here nodes 0, 1, A, B, C, D are fully validated and nodes
```

Comment:
Can we have a slightly more complicated one (fully validated nodes) here, 
like having more than one entry in synced_tips? 
Ex: 0,1,A,B,C,G,H,I

Reply:
That chain wouldn't be a valid synced tips